### ISSUE TITLE
MissingTraversableTypeHintSpecification does not take into account type aliases

### ISSUE THREAD

**Issue by marcospassos** [slevomat/coding-standard#1453]
Here's an example that illustrates the problem:

```php
    /**
     * @param WebEvaluationPayloadData $data
     */
    private function createEvaluationPayload(array $data): EvaluationPayload
    {
        return new EvaluationPayload(
            $data['query'],
            $this->createEvaluationContext($data['context'] ?? []),
        );
    }
```

The shape of the `$data` is pretty complex and composed of other subtypes used downstream. However, the `MissingTraversableTypeHintSpecification` reports it as a missing traversable type. 

For cases where the typed parameter isn't an array or iterable (from the sniff POV), I believe it should not report any error.

**Comment by kukulich** [comment]
@marcospassos How is `WebEvaluationPayloadData` defined?

**Comment by marcospassos** [comment]
Hi @kukulich! It's defined in the `phpstan.neon`:

```neon
includes:
    - vendor/phpstan/phpstan/conf/bleedingEdge.neon

parameters:
    typeAliases:
        PageData: 'array{title: string, url: string, referrer?: string}'
        CampaignData: 'array{name?: string, source?: string, medium?: string, content?: string, term?: string}'
        WebEvaluationContextData: 'array{timeZone?: string, campaign?: CampaignData, page?: PageData, attributes?: array<string, mixed>}'
        WebEvaluationPayloadData: 'array{query: string, context?: WebEvaluationContextData}'
```

**Comment by kukulich** [comment]
PHPCS cannot read this. You can use sniff configuration and `WebEvaluationPayloadData` mark as traversable.

**Comment by marcospassos** [comment]
I've been using this sniffer for a long time, and I didn't know that until now. Is it documented somewhere?

**Comment by mabar** [comment]
If you use phpstan/psalm then just drop the rule, static analysis handles these checks better and also does not require you to define it if it is already defined in interface/parent method

> Is it documented somewhere?

Only current file is read by PHPCS, don't expect anything that requires informations from other files to work
